inginious/frontend/pages/course_admin/task_list.py

```diff
@@ -2,89 +2,88 @@
 #
 # This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
 # more information about the licensing of this file.
-
+import json
+import logging
 from collections import OrderedDict
 
+import bson
 import web
 
-from inginious.frontend.pages.course_admin.utils import make_csv, INGIniousAdminPage
-
+from inginious.common.toc import check_toc
+from inginious.frontend.pages.course_admin.utils import INGIniousAdminPage
 
 class CourseTaskListPage(INGIniousAdminPage):
     """ List informations about all tasks """
 
     def GET_AUTH(self, courseid):  # pylint: disable=arguments-differ
         """ GET request """
-        course, __ = self.get_course_and_check_rights(courseid)
+        course, __ = self.get_course_and_check_rights(courseid, allow_all_staff=False)
         return self.page(course)
 
     def POST_AUTH(self, courseid):  # pylint: disable=arguments-differ
         """ POST request """
-        course, __ = self.get_course_and_check_rights(courseid)
-        data = web.input(task=[])
+        course, __ = self.get_course_and_check_rights(courseid, allow_all_staff=False)
 
-        if "task" in data:
-            # Change tasks order
-            for index, taskid in enumerate(data["task"]):
-                try:
-                    task = self.task_factory.get_task_descriptor_content(courseid, taskid)
-                    task["order"] = index
-                    self.task_factory.update_task_descriptor_content(courseid, taskid, task)
-                except:
-                    pass
+        errors = []
+        user_input = web.input()
+        try:
+            new_toc = json.loads(user_input["course_structure"])
+            valid, message = check_toc(new_toc)
+            if valid:
+                self.course_factory.update_course_descriptor_element(courseid, 'toc', new_toc)
+                course, __ = self.get_course_and_check_rights(courseid, allow_all_staff=False)  # don't forget to reload the modified course
+            else:
+                errors.append("Invalid table of content: " + message)
```

Comment:
Do not forget to wrap strings with ``_()``  to allow their translations

Reply:
Oups I forgot when I changed the errors. It's done